### PR TITLE
lots of documentation updates

### PR DESCRIPTION
**Pull request by larsks**
This patch adds inline documentation to all (well, most) of the
`defaults/main.yml` files in our Ansible roles.  It also includes
tools for generating `README.md` by extracting this documentation and
combining it with static content.

The tools and static content are located in the `docs/` directory.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,8 +41,25 @@ defines your hosts and maps them to host groups declared in
...
-example, I have a file called `config.yml` that contains:
+There are two high-level groups that can be used to control service
+placement:
+
+- `am_hosts`: The playbooks will install availability monitoring
+  software (Sensu, Uchiwa, and support services) on these hosts.
+
+- `logging_hosts`: The playbooks will install the centralized logging
+  stack (Elasticsearch, Kibana, Fluentd) on these hosts.
+
```
Comment (paramite) [inline README.md:52]: We are not going to document pm_hosts? I mean performance monitoring hosts deployment should be functional right? It does not matter we don't have analogical patch in TripleO, does it?

File: README.md [hunk 2]
```diff
@@ -30,19 +72,867 @@ example, I have a file called `config.yml` that contains:
...
+    Path to the Apache configuration snippet for the Uchiwa proxy.
+
+- `uchiwa_path` (default: `"/uchiwa"`)
+
+    URL path at which to host Uchiwa.
+
+
+## Chrony
+
+Installs and configures and NTP client ([Chrony][]) to ensure that the
```
Comment (paramite) [inline README.md]: "configures a NTP" perhaps? Though I'm not native, so maybe "an NTP".

**Review by paramite — APPROVED** [review]
+1 for this, we can complete performance monitoring documentation in other PRs

File: README.md (hunk 2 above)
Comment (larsks) [reply README.md]: Should be "an NTP client", yes.  Good catch.

File: README.md (hunk 1 above)
Comment (larsks) [reply README.md:52]: I wasn't going to document `pm_hosts` until it was actually functional.

**Pull request merged**